inversion-cosmosdb/src/main/java/io/inversion/cosmosdb/CosmosDb.java

```diff
@@ -151,7 +151,31 @@ public String upsertRow(Collection collection, Map<String, Object> row) throws A
             if (id == null) {
                 id = collection.encodeResourceKey(row);
                 if (id == null)
+                {
+                    // this is the first potential problem, where we couldn't get the primary index for some reason
+                    Index index = collection.getPrimaryIndex();
+                    if(index == null) {
+                        log.warn("INVERSION UNABLE TO DETERMINE PRIMARY INDEX");
+                        collection.getIndexes().forEach(cIndex -> {
+                            log.info("    {}", cIndex.toString());
+                        });
+                    }
+                    else {
+                        // this is the second potential problem, where we found the index but couldn't construct it
+                        // either it's the wrong index being chosen, or the row, by the time it gets here, doesn't
+                        // have the data required to construct the thing
+                        log.info("INVERSION USING INDEX [{]]", index.toString());
+                        if(row == null)
+                            log.warn("BUT THE ROW IS NULL!");
+                        else if(row.size() == 0)
+                            log.warn("BUT THE ROW IS EMPTY!");
+                        else
+                            index.getColumnNames().forEach(colName -> {
+                                log.info("INVERSION ROW MAP KEY [{}], VALUE [{}]", colName, row.get(colName));
```

Comment:
Can you change this to keep the data out of the logs.  This needs to be done lots of places but it could inadvertently cause security problems for some users.  Let's just log the name of the null fields(s).

Reply:
Yah, good catch.